### PR TITLE
Add index.json to manifest

### PR DESCRIPTION
**Pull request by qinezh**
@vwxyzh @superyyrrzz @hellosnow @ansyral @chenkennt

### CODE REVIEW

**Comment by chenkennt** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs [hunk 1]
```diff
@@ -79,7 +80,20 @@ where output.Key.Equals(".html", StringComparison.OrdinalIgnoreCase)
...
-            // TODO: add index.json to manifest
+
+            // add index.json to mainfest as resource file
+            var manifestItem = new ManifestItem
+            {
+                DocumentType = "Resource",
+                Metadata = new Dictionary<string, object>(),
+                OutputFiles = new Dictionary<string, OutputFileInfo>()
+            };
+            manifestItem.OutputFiles.Add("resource", new OutputFileInfo
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs]: "resource"? might not work @herohua

File: src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs [hunk 2]
```diff
@@ -79,7 +80,20 @@ where output.Key.Equals(".html", StringComparison.OrdinalIgnoreCase)
                 }
             }
             JsonUtility.Serialize(indexDataFilePath, indexData, Formatting.Indented);
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs]: >Serialize [](start = 24, length = 9)

conflict when having multiple version?

File: src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs (hunk 1 above)
Comment (herohua) [inline src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs]: the same with pictures, should be okay

File: src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs (hunk 2 above)
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs]: `index.json` should be saved in different folder. e.g. `_site/index.json`, `_site/v1.0/index.json`, `_site/v2.0/index.json`

**Comment by vwxyzh** [comment]
👍

**Pull request merged**